### PR TITLE
Issue #371 Fixing fmt and fmtcheck targets

### PR DESCRIPTION
**Pull request by hferentschik**
Addresses issue #371

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,8 +30,8 @@ LDFLAGS := -X $(REPOPATH)/pkg/version.version=$(VERSION) \
 	-X $(REPOPATH)/pkg/version.openshiftVersion=$(OPENSHIFT_VERSION) -s -w -extldflags '-static'
 
 GOFILES := go list  -f '{{join .Deps "\n"}}' ./cmd/minikube/ | grep $(REPOPATH) | xargs go list -f '{{ range $$file := .GoFiles }} {{$$.Dir}}/{{$$file}}{{"\n"}}{{end}}'
-GOFILES_NO_VENDOR := $(GOFILES) | grep -v /vendor
 PACKAGES := go list ./... | grep -v /vendor
+SOURCE_DIRS = cmd pkg
```
Comment (praveenkumar) [inline Makefile]: 👍

File: Makefile [hunk 2]
```diff
@@ -92,8 +92,8 @@ integration: $(BUILD_DIR)/$(GOOS)-$(GOARCH)/minishift$(IS_EXE)
 
 .PHONY: fmt
 fmt:
-	@gofmt -l -w $(shell $(GOFILES_NO_VENDOR))
+	@gofmt -l -w $(SOURCE_DIRS)
 
 .PHONY: fmtcheck
 fmtcheck:
-	@test -z $(shell gofmt -l -s $(shell $(GOFILES_NO_VENDOR)) | tee /dev/stderr) || echo "[WARN] Fix formatting issues with 'make fmt'"
+	@gofmt -l -s $(SOURCE_DIRS) | read; if [ $$? == 0 ]; then echo "gofmt check failed for:"; gofmt -l -s $(SOURCE_DIRS); exit 1; fi
```
Comment (praveenkumar) [inline Makefile]: yeah this will at-least fail the CI instead just warning.

**Comment by praveenkumar** [comment]
Looks like other than Makefile changes are just correcting gofmt warnings. LGTM 👍

File: Makefile (hunk 1 above)
Comment (budhrg) [inline Makefile]: @hferentschik , I think `test` directory should be included. It contains integration tests.

Comment (budhrg) [inline Makefile]: @hferentschik , I think `test` directory should be included. It contains integration tests.

Comment (praveenkumar) [inline Makefile]: @budhrg Agree it does contain integration test cases.

Comment (hferentschik) [inline Makefile]: Sure, I initially thought we wait until we have the new integration test framework, but I guess we might as well add this now

**Comment by hferentschik** [comment]
Added _test_ directory and merged

**Pull request merged**